integration-commons/src/main/java/org/symphonyoss/integration/healthcheck/IntegrationHealthManager.java

```diff
@@ -16,33 +16,93 @@
 import static org.symphonyoss.integration.model.healthcheck.IntegrationFlags.ValueEnum.OK;
 
+import com.symphony.api.pod.model.V1Configuration;
+import com.symphony.logging.ISymphonyLogger;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import org.apache.commons.lang3.StringUtils;
+import org.glassfish.jersey.client.ClientConfig;
+import org.glassfish.jersey.client.ClientProperties;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
 import org.symphonyoss.integration.IntegrationStatus;
+import org.symphonyoss.integration.authentication.AuthenticationProxy;
+import org.symphonyoss.integration.logging.IntegrationBridgeCloudLoggerFactory;
+import org.symphonyoss.integration.model.healthcheck.IntegrationConfigurator;
 import org.symphonyoss.integration.model.healthcheck.IntegrationFlags;
 import org.symphonyoss.integration.model.healthcheck.IntegrationHealth;
+import org.symphonyoss.integration.model.yaml.Application;
+import org.symphonyoss.integration.model.yaml.IntegrationBridge;
+import org.symphonyoss.integration.model.yaml.IntegrationProperties;
 
 import java.text.SimpleDateFormat;
 import java.util.Arrays;
 import java.util.Date;
 import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.Response;
 
 /**
  * Maintains the state of an individual Integration's health ({@link IntegrationHealth}), such as Jira, Github, etc.
  * Usually part of the basic attributes owned by any Integration on Integration Bridge.
  * Created by rsanchez on 03/08/16.
  */
+@Component
+@Scope(value = ConfigurableBeanFactory.SCOPE_PROTOTYPE)
 public class IntegrationHealthManager {
 
+  private static final ISymphonyLogger LOG =
+      IntegrationBridgeCloudLoggerFactory.getLogger(IntegrationHealthManager.class);
+
   private static final String SUCCESS = "Success";
 
+  private static final String OUT_OF_SERVICE = "Integration Out of Service. Please, check the flags";
+
   private static final String TIMESTAMP_FORMAT = "yyyy-MM-dd'T'HH:mm:ss'Z'Z";
 
+  private static final String APPS_CONTEXT = "apps";
+
+  private static final String APP_CONTROLLER_PAGE = "controller.html";
+
+  private static final String APP_ICON_IMAGE = "img/appstore-logo.png";
+
   private IntegrationHealth health = new IntegrationHealth();
 
   private IntegrationFlags flags = new IntegrationFlags();
 
+  @Autowired
+  private AuthenticationProxy authenticationProxy;
+
+  @Autowired
+  private IntegrationProperties properties;
+
+  /**
+   * Integration settings
+   */
+  private V1Configuration configuration;
```

Comment:
It still should not be used as a field in one of the IB classes.  Copy the data to a IB specific class.